chat-plugins/info.js

```diff
@@ -531,8 +531,12 @@ var commands = exports.commands = {
 						if (!template.learnset) template = Tools.getTemplate(template.baseSpecies);
 						if (!template.learnset) continue;
 						var fullLearnset = template.learnset;
-						while (template.prevo) {
-							template = Tools.getTemplate(template.prevo);
+						while (template.prevo || (template.eventPokemon && template.baseSpecies !== template.species)) {
+							if (template.prevo) {
+								template = Tools.getTemplate(template.prevo);
+							} else {
+								template = Tools.getTemplate(template.baseSpecies);
+							}
 							for (var move in template.learnset) {
 								if (!fullLearnset[move]) fullLearnset[move] = template.learnset[move];
```

Comment:
Could we deep clone the templates [here][1] and [here][2]? 

(At the risk of answering my own question, I imagine this gobbles up quite an unacceptable bit of memory.)

  [1]: https://github.com/gamut-was-taken/Pokemon-Showdown/blob/dexsearch/chat-plugins/info.js#L467
  [2]: https://github.com/gamut-was-taken/Pokemon-Showdown/blob/dexsearch/chat-plugins/info.js#L531